cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/SubqueryExpressionsTest.scala

```diff
@@ -0,0 +1,250 @@
+          ResultAssertions(r => {
+            r.toList should equal(List(Map("name" -> "Andy"), Map("name" -> "Peter")))
+          })) {
+          resultTable()
+        }
+      }
+      section("`EXISTS` subquery with `WHERE` clause", "existential-subquery-with-where") {
+        p(
+          """A `WHERE` clause can be used in conjunction to the `MATCH`.
+            #Variables introduced by the `MATCH` clause and the outside scope can be used in this scope.""".stripMargin('#'))
+        query(
+          """MATCH (person:Person)
+            #WHERE EXISTS {
+            #  MATCH (person)-[:HAS_DOG]->(dog:Dog)
+            #  WHERE person.name = dog.name
+            #}
+            #RETURN person.name AS name""".stripMargin('#'),
+          ResultAssertions(r => {
+            r.toList should equal(List(Map("name" -> "Andy")))
+          })) {
+          resultTable()
+        }
+      }
+      section("Nesting `EXISTS` subqueries", "existential-subquery-nesting") {
+        p(
+          """`EXISTS` subqueries can be nested like the following example shows.
+            #The nesting also affects the scopes.
+            #That means that it is possible to access all variables from inside the subquery which are either from the outside scope or defined in the very same subquery.""".stripMargin('#'))
+        query(
+          """MATCH (person:Person)
+            #WHERE EXISTS {
+            #  MATCH (person)-[:HAS_DOG]->(dog:Dog)
+            #  WHERE EXISTS {
+            #    MATCH (dog)-[:HAS_TOY]->(toy:Toy)
+            #    WHERE toy.name = 'Banana'
+            #  }
+            #}
+            #RETURN person.name AS name""".stripMargin('#'),
+          ResultAssertions(r => {
+            r.toList should equal(List(Map("name" -> "Peter")))
+          })) {
+          resultTable()
+        }
+      }
+      section("`EXISTS` subquery outside of a `WHERE` clause", "existential-subquery-outside-where") {
+        p(
+          """`EXISTS` subquery expressions can appear anywhere that an expression is valid.
+            |Here the result if the subquery can find the given pattern is returned.""".stripMargin)
+        query(
+          """MATCH (person:Person)
+            #RETURN person.name AS name, EXISTS {
+            #  MATCH (person)-[:HAS_DOG]->(:Dog)
+            #} AS hasDog""".stripMargin('#'),
+          ResultAssertions(r => {
+            r.toList should equal(List(
+              Map("name" -> "Andy", "hasDog" -> true),
+              Map("name" -> "Timothy", "hasDog" -> false),
+              Map("name" -> "Peter", "hasDog" -> true)
+            ))
+          })) {
+          resultTable()
+        }
+      }
+    }
+
+    section("`COUNT` subqueries", "count-subqueries") {
+      p("A `COUNT` subquery expression can be used to to count the number of results of the subquery.")
+      functionWithCypherStyleFormatting(
+        """COUNT {
+          #  [Pattern]
```

Comment:
As of what you said, this should read:
```suggestion
          # MATCH [Pattern]
```